### PR TITLE
fix/miscellaneous

### PR DESCRIPTION
**Pull request by edwinmp**
- Validation on aggregate, transform & join
- Reduce bundle size with dynamic imports
- Process feedback from user testing

### CODE REVIEW

**Comment by k8hughes** [comment]
@edwinmp is this ready to be merged?

File: frontend/src/pages/QueryData/QueryData.tsx [hunk 1]
```diff
@@ -113,14 +125,17 @@ class QueryData extends React.Component<QueryDataProps> {
...
   private exportToCSV = () => {
+    this.setState({ exporting: true });
     const data = this.props.page.getIn([ 'data', 'results' ]) as List<OperationDataMap>;
     if (data.count()) {
-      const csv = encodeURI(`data:text/csv;charset=utf-8,${unparse(data.toJS())}`);
       const filename = 'export.csv';
       const link = document.createElement('a');
-      link.setAttribute('href', csv);
+      const csvData = new Blob([ unparse(data.toJS()) ], { type: 'text/csv' });
+      const url = URL.createObjectURL(csvData);
```
Comment (akmiller01) [inline frontend/src/pages/QueryData/QueryData.tsx:134]: @edwinmp is it possible to add a UTF8 BOM to the beginning of this file? Excel cannot natively handle UTF8 without it.

Should be something like 
```
\ufeff
```
Via https://stackoverflow.com/a/17879474

Comment (edwinmp) [reply frontend/src/pages/QueryData/QueryData.tsx:134]: Done!

**Comment by edwinmp** [comment]
@akmiller01, approve when you're done reviewing.

**Comment by akmiller01** [comment]
Hey @edwinmp this didn't work until I `npm install node-sass`. Could you add that to the package?

**Comment by akmiller01** [comment]
Also getting random page-breaking errors :( 
image
nvm, clearing cache fixes it.

**Comment by edwinmp** [comment]
> Hey @edwinmp this didn't work until I `npm install node-sass`. Could you add that to the package?

Deleted my node_modules and reinstalled all the dependencies, then ran the app ... no issues or requirements for `node-sass` ... expected as much since `node-sass` is already a dependency in `package.json`

**Comment by edwinmp** [comment]
> Also getting random page-breaking errors :(
> image
> nvm, clearing cache fixes it.

Couldn't replicate this.

**Review by akmiller01 — APPROVED** [review]
Working well. Still not sure why the staging server needed node-sass installed manually :man_shrugging:

**Pull request merged**